packages/mendel-pipeline/src/step/waiter.js

```diff
@@ -0,0 +1,20 @@
+const BaseStep = require('./step');
+
+class Waiter extends BaseStep {
+    /**
+     * @param {MendelCache} toolset.cache
+     */
+    constructor({cache}) {
+        super();
+        this.waitCount = 0;
+        this.cache = cache;
+    }
+
+    perform() {
+        this.waitCount++;
+        if (this.cache.size() > this.waitCount) return;
+        this.cache.entries().forEach(({id}) => this.emit('done', {entryId: id}));
+    }
+}
```

Comment:
That is a cleaver step implementation.

It creates a virtual bottleneck tho. For instance, before this step, the consumer/generator/outlet/middleware side cache was being wormed up in parallel.

`// TODO: Once type refactor is done and plan is part of entry instance, it will be safe to add optimization here to emit entries without GST right away before wait condition is met`